Wrappers/Python/cil/framework/framework.py

```diff
@@ -1822,32 +1880,30 @@ def __init__ (self, array, deep_copy=True, dimension_labels=None,
         else:
             raise TypeError('Array must be NumpyArray, passed {0}'\
                             .format(type(array)))
-        
+
+        #Don't set for derived classes
+        if type(self) is DataContainer:
+            self.dimension_labels = dimension_labels
+
         # finally copy the geometry
         if 'geometry' in kwargs.keys():
             self.geometry = kwargs['geometry']
-        else:
-            # assume it is parallel beam
-            pass
         
     def get_dimension_size(self, dimension_label):
-        if dimension_label in self.dimension_labels.values():
-            acq_size = -1
-            for k,v in self.dimension_labels.items():
-                if v == dimension_label:
-                    acq_size = self.shape[k]
-            return acq_size
+
+        if dimension_label in self.dimension_labels:
+            i = self.dimension_labels.index(dimension_label)
+            return self.shape[i]
         else:
             raise ValueError('Unknown dimension {0}. Should be one of {1}'.format(dimension_label,
                              self.dimension_labels))
     def get_dimension_axis(self, dimension_label):
-        if dimension_label in self.dimension_labels.values():
-            for k,v in self.dimension_labels.items():
-                if v == dimension_label:
-                    return k
+
+        if dimension_label in self.dimension_labels:
+            return self.dimension_labels.index(dimension_label)
         else:
             raise ValueError('Unknown dimension {0}. Should be one of {1}'.format(dimension_label,
-                             self.dimension_labels.values()))
+                             self.dimension_labels))
                         
     def as_array(self, dimensions=None):
```

Comment:
remove the possibility to call `as_array` to slice, which is provided by the new `get_slice`